src/App.jsx

```diff
@@ -14,33 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const initialGoods = goodsFromServer;
+  const [goods, setGoods] = useState([...initialGoods]);
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const updateGoods = () => {
+    const newGoods = [...initialGoods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    if (sortBy === 'alphabet') {
+      newGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortBy === 'length') {
+      newGoods.sort((a, b) => a.length - b.length);
+    }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+    if (isReversed) {
+      newGoods.reverse();
+    }
+
+    setGoods(newGoods);
+  };
+
+  useEffect(() => {
+    updateGoods();
+  }, [sortBy, isReversed]);
+
+  const handleSortAlphabetically = () => {
+    setSortBy('alphabet');
+  };
+
+  const handleSortByLength = () => {
+    setSortBy('length');
+  };
+
+  const handleReverse = () => {
+    setIsReversed(!isReversed);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReset = () => {
+    setSortBy('');
+    setIsReversed(false);
+    setGoods(initialGoods);
+  };
+
+  const isInitialOrder =
+    sortBy === '' &&
+    !isReversed &&
+    goods.every((good, index) => good === initialGoods[index]);
```

Comment:
```suggestion
  const isInitialOrder = !sortBy && !isReversed
```
You don't need to compare with an empty string, just check if value exists